### PR TITLE
Gif article comment

### PR DESCRIPTION
**Pull request by aiyus2020**

### CODE REVIEW

File: controllers/commentController.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+      //query to insert into the gif comment table
+      const myGifComment = await client.query(gifCommentQuery, [
+        userId,
+        id,
+        comment,
+      ]);
+
+      //query to access the gif table and so as to get the its title
+      const getGifTitle = await client.query(findId, [id]);
```
Comment (courage173) [inline controllers/commentController.js:24]: no, there is no point adding this to what you are returning. please remove it

File: controllers/commentController.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+      res.json({
+        status: "success",
+        data: {
+          id: myGifComment.rows[0].id,
+          message: "comment successfully created",
```
Comment (courage173) [inline controllers/commentController.js:30]: the message should not be inside the data, you should just return the created comment

File: controllers/commentController.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+      //query to insert into the gif comment table
+      const myArticleComment = await client.query(articleCommentQuery, [
+        userId,
+        id,
+        comment,
+      ]);
+
+      //query to access the gif table and so as to get the its title
+      const getArticle = await client.query(findArticleId, [id]);
```
Comment (courage173) [inline controllers/commentController.js:54]: same with this

File: controllers/gifsController.js [hunk 4]
```diff
@@ -57,6 +57,7 @@ class GifsController {
       res.json({
         status: "success",
         data: {
+          upload: upload.rows[0].cloud_public_id,
```
Comment (courage173) [inline controllers/gifsController.js:60]: why should you add this after deleting?? the message is enough

File: models/dbMigration.js [hunk 5]
```diff
@@ -52,6 +52,23 @@ async function dbMigration() {
         updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
       )`;
     await client.query(articles);
+
+    const articleComment = `CREATE TABLE IF NOT EXISTS article_comments (
```
Comment (courage173) [inline models/dbMigration.js:56]: if you have this why do you still need the db sql

File: queries/commentQuery.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+const client = require("../models/dbMigration");
+const gifCommentQuery =
+  "INSERT INTO gif_comments (user_id, gif_id, comment) VALUES ($1, $2, $3) RETURNING *";
+const articleCommentQuery =
+  "INSERT INTO article_comments (user_id, article_id, comment) VALUES ($1, $2, $3) RETURNING *";
+//find by id
+const findArticleId = "SELECT * FROM articles WHERE id=$1";
```
Comment (courage173) [inline queries/commentQuery.js:7]: this should be findArticleById

**Review by courage173 — CHANGES_REQUESTED** [review]

File: controllers/commentController.js (hunk 1 above)
Comment (aiyus2020) [reply controllers/commentController.js:24]: gifs

I did it to assess the gifs title in the upload gifs table,
so I can return it in the response.

File: controllers/commentController.js (hunk 2 above)
Comment (aiyus2020) [reply controllers/commentController.js:30]: gifs
The message was returned in the template shown above

File: controllers/commentController.js (hunk 3 above)
Comment (aiyus2020) [reply controllers/commentController.js:54]: art

I did it to assess the article title in the article table,
so I can return it in the response.

File: models/dbMigration.js (hunk 5 above)
Comment (aiyus2020) [reply models/dbMigration.js:56]: I don't really need it, was just writing it initially to practice SQL. will delete it now

File: queries/commentQuery.js (hunk 6 above)
Comment (aiyus2020) [reply queries/commentQuery.js:7]: noted, and correction was effected accordingly. sir

File: controllers/commentController.js (hunk 1 above)
Comment (courage173) [reply controllers/commentController.js:24]: okay. but please remove it. no point making that extra call

Comment (aiyus2020) [reply controllers/commentController.js:24]: noted sire.

**Review by courage173 — CHANGES_REQUESTED** [review]

File: routes/commentRoute.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const CommentController = require("../controllers/commentController");
+const router = express.Router();
+const authenticate = require("../middleware/authmiddleware");
+
+router.post(
```
[the remark this answers is not in the document]
Comment (aiyus2020) [reply routes/commentRoute.js:8]: Added validation for articles, article comments and gif comments respectively sir. 👌

File: controllers/commentController.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+      const { id } = req.params; //get/access the gif id
+      const { comment } = req.body; // get/access the comment
+
+      //query to insert into the gif comment table
+      const myGifComment = await client.query(gifCommentQuery, [
+        userId,
+        id,
+        comment,
+      ]);
+      if (myGifComment.rows.comment.le == 0) {
```
Comment (courage173) [inline controllers/commentController.js:19]: what is this please?

**Review by courage173 — CHANGES_REQUESTED** [review]
please remove that code you added and you can merge

File: controllers/commentController.js (hunk 8 above)
Comment (aiyus2020) [reply controllers/commentController.js:19]: it was a mistake.
will remove it

**Pull request merged**